### ISSUE TITLE
A11y: [calcite-color-picker] - focus should be set on the color field

### ISSUE THREAD

**Issue by hccampos** [Esri/calcite-components#2267]
### Summary
Currently, calling `setFocus` on a `<calcite-color-picker>` sets the focus on the `hexInputNode` but that the first focusable element in the component is the color field. 

### Actual Behavior
- Calling `setFocus` on a `<calcite-color-picker>` sets the focus on the `hexInputNode`.
- Furthermore, if `.setFocus()` is called immediately after creating the color picker, the hex input won't be initialized and `.setFocus()` will be no op.

### Expected Behavior
- The first focusable element in the `<calcite-color-picker>` is the main color field handle, so it should be focused when calling `.setFocus()`.
- If `.setFocus()` is called before the internal elements are present, the color picker should remember this and set focus when possible. Alternatively, the component could provide a `whenFocusable()` method (or similar).

### Reproduction Steps

1. Create a `<calcite-color-picker>`.
2. Call `element.setFocus()`.
3. Note that the focus is not set right away because the hex input isn't ready immediately.
4. Wait for the hex input to be ready and call `.setFocus()` again.
5. Note that the focus is set, but on the hex input which isn't the first focusable element of the component.

**Comment by jcfranco** [comment]
Closing as self-verified with the following: https://codepen.io/jcfranco/pen/YzVZdLj?editors=1000
